src/api/entities/Identity/IdentityAuthorizations.ts

```diff
@@ -51,4 +53,41 @@ export class IdentityAuthorizations extends Authorizations<Identity> {
       next,
     };
   }
+
+  /**
+   * Retrieve a single Authorization Request targeting or issued by this Identity by its ID
+   *
+   * `authorizations` storage only returns results for the authorization target, so `authorizationsGiven` needs to be queried first to find the relevant target if present
+   *
+   * @throws if there is no Authorization Request with the passed ID targeting or issued by this Identity
+   */
+  public override async getOne(args: { id: BigNumber }): Promise<AuthorizationRequest> {
+    const {
+      context,
+      parent: { did },
+      context: {
+        polymeshApi: {
+          query: { identity },
+        },
+      },
+    } = this;
+
+    const { id } = args;
+
+    const rawId = bigNumberToU64(id, context);
+
+    const targetSignatory = await identity.authorizationsGiven(
+      stringToIdentityId(did, context),
+      rawId
+    );
+
+    if (!targetSignatory.isEmpty) {
+      const auth = await identity.authorizations(targetSignatory, rawId);
+      const target = signatoryToSignerValue(targetSignatory);
+
+      return this.createAuthorizationRequests([{ auth: auth.unwrap(), target }])[0];
+    }
+
+    return super.getOne({ id });
```

Comment:
We can get a performance boost (if we want) by calling this before awaiting `identity.authorizationsGiven` and saving it into a variable so that it runs in parallel. You would probably have to call `defusePromise` on it in case it errors out before you return it

So

```
/*
 * We're using `defusePromise` here because if the id corresponds to a sent auth, we don't
 *   care about the error, and if it doesn't, then we return the promise and the error will be
 *   handled by the caller
 */
const gettingReceivedAuth = defusePromise(super.getOne({ id }));

// ... logic, fetch the sent promise and stuff

return gettingReceivedAuth;
```

Just a suggestion, if you think it hurts readability too much ignore it. We can worry about performance later on

Reply:
I liked the approach and I don't think it hurts readability. The comments make it crystal clear.